src/cli/cli.cpp

```diff
@@ -276,11 +276,33 @@ template <> otError Interpreter::Process<Cmd("version")>(Arg aArgs[])
 
 template <> otError Interpreter::Process<Cmd("reset")>(Arg aArgs[])
 {
-    OT_UNUSED_VARIABLE(aArgs);
+    otError error = OT_ERROR_NONE;
 
-    otInstanceReset(GetInstancePtr());
+    if (aArgs[0].IsEmpty())
+    {
+        otInstanceReset(GetInstancePtr());
+    }
 
-    return OT_ERROR_NONE;
+#if OPENTHREAD_CONFIG_PLATFORM_BOOTLOADER_MODE_ENABLE
+    /**
+     * @cli reset bootloader
+     * @code
+     * reset bootloader
+     * @endcode
+     * @par
+     * Resets platform to bootloader mode, if supported.
+     */
+    else if (aArgs[0] == "bootloader")
```

Comment:
Should we add a cli test to `tests/scripts/expect/cli-reset.exp`?

Reply:
Good idea! Just added the test